metasrv/tests/it/grpc/metasrv_grpc_watch.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_base::tokio;
+use common_meta_api::KVApi;
+use common_meta_grpc::MetaGrpcClient;
+use common_meta_types::protobuf::event::SeqV;
+use common_meta_types::protobuf::watch_request::FilterType;
+use common_meta_types::protobuf::Event;
+use common_meta_types::protobuf::WatchRequest;
+use common_meta_types::MatchSeq;
+use common_meta_types::Operation;
+use common_meta_types::UpsertKVAction;
+
+use crate::init_meta_ut;
+
+async fn upsert_kv_client_main(addr: String, updates: Vec<UpsertKVAction>) -> anyhow::Result<()> {
+    let client = MetaGrpcClient::try_create(addr.as_str(), "root", "xxx", None, None).await?;
+
+    // update some kv
+    for update in updates.iter() {
+        let _ = client.upsert_kv(update.clone()).await;
+    }
+
+    Ok(())
+}
+
+async fn test_watch_main(
+    addr: String,
+    watch: WatchRequest,
+    mut watch_events: Vec<Event>,
+    updates: Vec<UpsertKVAction>,
+) -> anyhow::Result<()> {
+    let client = MetaGrpcClient::try_create(addr.as_str(), "root", "xxx", None, None).await?;
+
+    let mut grpc_client = client.make_client().await?;
+
+    let request = tonic::Request::new(watch);
+
+    let mut client_stream = grpc_client.watch(request).await?.into_inner();
+
+    let _h = tokio::spawn(upsert_kv_client_main(addr.clone(), updates));
+
+    loop {
+        tokio::select! {
+            resp = client_stream.message() => {
+                if let Ok(Some(resp)) = resp {
+                    if let Some(event) = resp.event {
+                        assert!(!watch_events.is_empty());
+
+                        assert_eq!(watch_events.get(0), Some(&event));
+                        watch_events.remove(0);
```

Comment:
This should be enough to guarantee what you expect.
```suggestion
                        assert_eq!(watch_events.remove(0), Some(&event));
```

Reply:
not a big deal